### PR TITLE
Issue 16565 dotasset

### PR DESCRIPTION
**Pull request by jdotcms**
Adding a new base content type called dotAsset, which is basically a binary without path

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import com.dotcms.business.CloseDBIfOpened;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.portlets.workflows.business.WorkflowAPI;
+import com.dotmarketing.startup.StartupTask;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UUIDGenerator;
+
+import java.util.List;
+import java.util.Map;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Map'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+        // Inserts into Structure table the Language Variable Content Type
+        this.addDotAssetContentType();
+        // Gets the list of required fields for the Language Variable Content Type
+        this.addDotAssetFields();
+    }
+
+    private void addDotAssetFields() throws DotDataException {
+
+        final DotConnect dc = new DotConnect();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java]: Codacy Issue found: [Avoid variables with short names like dc](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.Set;
+import java.util.stream.Collectors;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.stream.Collectors'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        final List<Map<String, Object>> requiredFields = getDotAssetVariableFields();
+        int i = 0;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java]: Codacy Issue found: [Avoid variables with short names like i](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+     * @param fixed
+     * @param readOnly
+     * @param searchable
+     * @param unique
+     * @return The data required by a field.
+     */
+    private Map<String, Object> setFieldElements(String fieldName, String fieldType, String fieldRelationType,
+                    String fieldContentlet, boolean required, boolean listed, boolean indexed, String velocityVarName,
+                    int sortOrder, String values, String checkRegex, String hint, String defaultValue, boolean fixed,
+                    boolean readOnly, boolean searchable, boolean unique) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java]: Codacy Issue found: [Parameter 'unique' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+import com.dotcms.business.CloseDBIfOpened;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.portlets.workflows.business.WorkflowAPI;
+import com.dotmarketing.startup.StartupTask;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UUIDGenerator;
+
+import java.util.List;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.List'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+     * @param checkRegex
+     * @param hint
+     * @param defaultValue
+     * @param fixed
+     * @param readOnly
+     * @param searchable
+     * @param unique
+     * @return The data required by a field.
+     */
+    private Map<String, Object> setFieldElements(String fieldName, String fieldType, String fieldRelationType,
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java]: Codacy Issue found: [Parameter 'fieldType' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+                    String fieldContentlet, boolean required, boolean listed, boolean indexed, String velocityVarName,
+                    int sortOrder, String values, String checkRegex, String hint, String defaultValue, boolean fixed,
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java]: Codacy Issue found: [Parameter 'values' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.Map;
+import java.util.Set;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Set'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+     * @param hint
+     * @param defaultValue
+     * @param fixed
+     * @param readOnly
+     * @param searchable
+     * @param unique
+     * @return The data required by a field.
+     */
+    private Map<String, Object> setFieldElements(String fieldName, String fieldType, String fieldRelationType,
+                    String fieldContentlet, boolean required, boolean listed, boolean indexed, String velocityVarName,
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java]: Codacy Issue found: [Parameter 'velocityVarName' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.stream.Collectors'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java]: Codacy Issue found: [Avoid variables with short names like i](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java]: Codacy Issue found: [Parameter 'velocityVarName' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.List'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java]: Codacy Issue found: [Avoid variables with short names like dc](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Set'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java]: Codacy Issue found: [Parameter 'unique' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java]: Codacy Issue found: [Parameter 'fieldType' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05215AddSystemWorkflowToDotAssetContentType.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Map'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAsset.java]: Codacy Issue found: [Parameter 'values' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAssetTest.java [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+import org.junit.Test;
+
+import java.sql.SQLException;
+
+import static org.junit.Assert.assertTrue;
+
+public class Task05210CreateDefaultDotAssetTest {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAssetTest.java]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAssetTest.java [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+            } catch (DotDataException e) {
+                //Nah.
+            }
+        }
+
+        for (final String fieldInode : fieldInodes) {
+            try {
+                new DotConnect().setSQL("delete from inode where inode = ?")
+                        .addParam(fieldInode).loadResult();
+            } catch (DotDataException e) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05210CreateDefaultDotAssetTest.java]: Codacy Issue found: [Avoid empty catch blocks](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995431)

**Review by nollymar — COMMENTED** [review]
Did you run these tests using empty starter/empty DB?

**Review by erickgonzalez — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Pull request merged**